### PR TITLE
Update context in `RadioGroup`

### PR DESCRIPTION
**Pull request by jjenzz**

### CODE REVIEW

File: packages/react/radio-group/src/RadioGroup.tsx [hunk 1]
```diff
@@ -95,9 +81,18 @@ const RadioGroup = React.forwardRef((props, forwardedRef) => {
...
-    <RadioGroupContext.Provider value={context}>
+    <RadioGroupProvider
+      name={name}
+      value={value}
+      required={required}
+      rovingFocus={rovingFocus}
+      onValueChange={React.useCallback(
+        composeEventHandlers(handleValueChange, (event) => setValue(event.target.value)),
+        [handleValueChange]
+      )}
```
Comment (jjenzz) [inline packages/react/radio-group/src/RadioGroup.tsx:92]: This made me realise that we're passing the `event` to the consumer which doesn't seem to be inline with how we handle things elsewhere. I was expecting it to be `onValueChange(value: string): void;`. 

The docs say it has that signature but it actually doesn't 😬 

image

Shall I change it while I'm here?

Comment (benoitgrelard) [reply packages/react/radio-group/src/RadioGroup.tsx:92]: hmm weird! Let's tackle separately.

**Review by benoitgrelard — APPROVED** [review]

**Comment by jjenzz** [comment]
Before merging this, I'll wait for the context update PR to go in (#496) and then update this to remove the `React.useCallback`

**Comment by jjenzz** [comment]
@benoitgrelard Happy for me to leave this as is and merge now that we've decided to close #496 ?

**Review by benoitgrelard — APPROVED** [review]

**Pull request merged**